File: javascript/Customer.js
// javascript/Customer.js
export class Customer {
    constructor(name, balance) {
        this.name = name;
        this.balance = balance;
    }

    getName() {
        return this.name;
    }

    getBalance() {
        return this.balance;
    }

    deductBalance(amount) {
        if (amount > this.balance) {
            throw new Error("Insufficient balance");
        }
        this.balance -= amount;
    }

    hasBalance(amount) {
        return this.balance >= amount;
    }
}

File: csharp/Customer.cs
// csharp/Customer.cs
using System;

namespace ECommerce
{
    public class Customer
    {
        public string Name { get; private set; }
        public double Balance { get; private set; }

        public Customer(string name, double balance)
        {
            Name = name;
            Balance = balance;
        }

        public void DeductBalance(double amount)
        {
            if (amount > Balance)
            {
                throw new ArgumentException("Insufficient balance");
            }
            Balance -= amount;
        }

        public bool HasBalance(double amount)
        {
            return Balance >= amount;
        }
    }
}
